src/realm/spec.hpp

```diff
@@ -273,28 +214,20 @@ inline Spec::Spec(Allocator& alloc) noexcept
 {
 }
 
-inline SubspecRef Spec::get_subtable_spec(size_t column_ndx) noexcept
-{
-    REALM_ASSERT(column_ndx < get_column_count());
-    REALM_ASSERT(get_column_type(column_ndx) == col_type_Table);
-    size_t subspec_ndx = get_subspec_ndx(column_ndx);
-    return SubspecRef(&m_subspecs, subspec_ndx);
-}
-
-inline ConstSubspecRef Spec::get_subtable_spec(size_t column_ndx) const noexcept
+inline Spec* Spec::get_subtable_spec(size_t column_ndx) noexcept
 {
     REALM_ASSERT(column_ndx < get_column_count());
     REALM_ASSERT(get_column_type(column_ndx) == col_type_Table);
     size_t subspec_ndx = get_subspec_ndx(column_ndx);
-    return ConstSubspecRef(&m_subspecs, subspec_ndx);
+    return get_subspec_by_ndx(subspec_ndx);
 }
 
-inline SubspecRef Spec::get_subspec_by_ndx(size_t subspec_ndx) noexcept
+inline const Spec* Spec::get_subtable_spec(size_t column_ndx) const noexcept
 {
-    return SubspecRef(&m_subspecs, subspec_ndx);
+    return const_cast<Spec*>(this)->get_subtable_spec(column_ndx);
```

Comment:
Does this method need to be const? It would be ideal to avoid the `const_cast` if possible because it actually could change the object if creating the subspec for the first time.